langchain/src/chat_models/openai.ts

```diff
@@ -355,6 +357,35 @@ export class ChatOpenAI extends BaseChatModel implements OpenAIInput {
     };
   }
 
+  getNumTokensFromMessages(messages: BaseChatMessage[]): {
+    totalCount: number;
+    countPerMessage: number[];
+  } {
+    let totalCount = 0;
+    let tokensPerMessage = 0;
+    let tokensPerName = 0;
+
+    // From: https://github.com/openai/openai-cookbook/blob/main/examples/How_to_format_inputs_to_ChatGPT_models.ipynb
+    if (["gpt-3.5-turbo", "gpt-3.5-turbo-0301"].includes(this.modelName)) {
+      tokensPerMessage = 4;
+      tokensPerName = -1;
+    } else if (["gpt-4", "gpt-4-0314"].includes(this.modelName)) {
+      tokensPerMessage = 3;
+      tokensPerName = 1;
+    }
+
+    const countPerMessage = messages.map((message) => {
+      const textCount = this.getNumTokens(message.text);
```

Comment:
Note this method (from the base class) needs to be fixed for this to be accurate. It is using the wrong tokeniser. The solution is probably to convert it to use tiktoken. This doesn't have to be done in this PR, just raising it as an FYI

Reply:
gotcha, good catch.

I can try to flip it over today if it's already a dependency